Move CESM SE reader file to monetio and add example files for CESM SE

Duseong opened the pull request:
I moved CESM SE reader file, and created a jupyter notebook example file with yaml file under docs/examples

Example output files were also stored in docs/examples/output/
Maybe we can delete example png files to save disk space. 

I will also create a pull request for monetio soon.

Duseong commented:
I checked where this CI build fail comes from, and it showed this error:

`Extension error:
Could not import extension myst_nb (exception: No module named 'myst_nb')`

it looks like this "myst_nb" extension is included in docs/conf.py

zmoon commented:
Hi @Duseong.

* The docs CI failure is because you added a docs example notebook in there but didn't add it to the index (or ignore/hide it)
  - edit `docs/index.rst` to get it in there
* Since you are using `debug: True` and assuming you pushed the executed notebook, the figures should be in there, so you don't need to include them separately
  - instead add a `.gitignore` like the idealized and WRF-Chem example output dirs have
  - and `git rm` the figure files

Sorry that adding new example nb is kind of complicated currently

zmoon commented:
Also we can add your example dataset to the tutorial module so that it can be loaded automatically. You can see [here](https://melodies-monet.readthedocs.io/en/develop/examples/tutorial-data.html) how it is supposed to work.

Duseong commented:
> Hi @Duseong.
> 
> * The docs CI failure is because you added a docs example notebook in there but didn't add it to the index (or ignore/hide it)
>   
>   * edit `docs/index.rst` to get it in there
> * Since you are using `debug: True` and assuming you pushed the executed notebook, the figures should be in there, so you don't need to include them separately (like you suggested)
>   
>   * instead add a `.gitignore` like the idealized and WRF-Chem example output dirs have
>   * and `git rm` the figure files
> 
> Sorry that adding new example nb is kind of complicated currently

Thanks always for your detailed explanations! I followed your steps, I hope it works

Duseong commented:
> Also we can add your example dataset name and location to the tutorial module so that it can be loaded automatically. You can see [here](https://melodies-monet.readthedocs.io/en/develop/examples/tutorial-data.html) how it is supposed to work.
> 
> Do we need to upload the SCRIP file to [here](https://csl.noaa.gov/groups/csl4/modeldata/melodies-monet/data/example_model_data/musica_example/) as well?

Ah that's right. Can I upload the file there directly? I have been asking Rebecca to upload sample files, maybe I will ask Rebecca again for help. 

For tutorial mode, I guess I need to modify these two files? tutorial.py and control_camchem_se.yaml

zmoon reviewed (commented):
important for the gitignore file to be called `.gitignore`, I'm seeing `.girignore`

Duseong commented:
> important for the gitignore file to be called `.gitignore`, I'm seeing `.girignore`

I am sorry, I fixed it.

Duseong commented:
Becky just shared the ftp site info with me. I just uploaded the scrip file to the ftp site. https://csl.noaa.gov/groups/csl4/modeldata/melodies-monet/data/example_model_data/musica_example/

zmoon commented:
> For tutorial mode, I guess I need to modify these two files? tutorial.py and control_camchem_se.yaml

So yes you just need to add in the dict in `tutorial.py` and then change the files in your control file

But also a handling for the SCRIP file will have to be added in driver similar to how I did it for the main model file

Duseong commented:
Could you let me know how can I get sha256 for new files? Actually I haven't heard of this, so I searched on the web and found some SHA256 has generator site: https://passwordsgenerator.net/sha256-hash-generator/

I just checked it using the existing datasets but it gave me a different sha256 value, so I am not sure how to generate sha256 for MELODIES-MONET.

zmoon commented:
> Could you let me know how can I get sha256 for new files?

On Linux, you should have the `sha256sum` command that you can use

Or, if you add it, but put `None` for the checksum, pooch will tell you the sha256 when it first downloads the file

Duseong commented:
Thanks @zmoon! I updated files for the tutorial module and tested it on Cheyenne, and confirmed that both CESM-SE and CESM-SE scrip files were downloaded successfully.

zmoon reviewed (commented):
Nice. I don't see modification to driver though to account for the SCRIP file, does it run with `scrip_file: example:camchem:se_scrip`?

Duseong commented:
`import melodies_monet as mm

mm.tutorial.example_ids
['wrfchem:racm_esrl',
 'wrfchem:racm_esrl_vcp',
 'airnow:2019-09',
 'airnow:2019-08',
 'airnow:2019-07',
 'camchem:fv',
 'camchem:se',
 'camchem:se_scrip']

fp = mm.tutorial.fetch_example("camchem:se")
fp2 = mm.tutorial.fetch_example("camchem:se_scrip")

ds = xr.open_dataset(fp)
ds2 = xr.open_dataset(fp2)`


This is what I tested following the tutorial docs page, did I miss something?

zmoon commented:
> This is what I tested following the tutorial docs page, did I miss something?

I meant your notebook, does it still run with that modification to the control file?

Duseong commented:
Oh, you meant to test the whole example notebook using it, I didn't, I will test it right away.

Duseong commented:
Thanks again, now finally I understand how this tutorial module works in MELODIES-MONET. I updated the driver.py and confirmed airnow_camchem_se.ipynb file is working with the tutorial module.

zmoon commented:
@Duseong did you build the docs locally to make sure you are happy with your notebook looks in there?

Duseong commented:
I added a reference in docs/examples/intro_examples.rst. 

I am not sure how to check it locally, but since I copied camchem.ipynb file and just changed related filenames, it should be ok. 

To locally check, do I have to do this command and find some generated document in some directory?
`sphinx-build -D execution_show_tb=True -b html -W docs docs/_build/html`

It actually raises error for several extensions, so I am currently installing extensions like myst_nb and sphinx_design

Duseong commented:
Oh, it looks like it generated html files in this directory: /docs/_build/html

Duseong commented:
I just checked html/examples/airnow_camchem_se.html. It looks ok, similar to camchem_fv example page. It's amazing to see that html documents are automatically generated from the jupyter notebook files! There are so many advanced systems that I am not aware of.

zmoon commented:
The `docs/environment-docs.yml` has the docs dependencies.

Probably we need a developer's guide section on how to add a new docs example notebook. At least we can use this PR thread as a basis

zmoon commented:
Ok and then final question, do you have any interest in adding a few sentences about the intended/desired interpretation of the results in your example and/or something about the study period (FIREX-AQ?), maybe even model configuration details?

zmoon commented:
> It's amazing to see that html documents are automatically generated from the jupyter notebook files! There are so many advanced systems that I am not aware of.

Yeah Sphinx and its ecosystem are very powerful for sure. [MyST-NB](https://myst-nb.readthedocs.io/en/latest/) that we use here comes to us from the Executable Books Project, which is also worth checking out if you haven't.

Duseong commented:
> Ok and then (hopefully) final question, do you have any interest in adding a few sentences about the intended/desired interpretation of the results in your example and/or something about the study period (FIREX-AQ?), maybe even model configuration details?

Yes, is there any place I can put in a few sentences? It's generated just for demonstration, so it shouldn't be used for any scientific research. But I would like to advertise where this sample example file comes from, Community Simulations (https://dashrepo.ucar.edu/dataset/256_tilmes.html).

zmoon commented:
> is there any place I can put in a few sentences?

I think near the top is good. For the WRF-Chem one I put in a few sentences from Becky there.

Duseong commented:
I added a couple of sentences at the top of the example file.

Duseong commented:
Thanks Zach for your time in reviewing these codes and giving a lot of helpful comments!
